Core/Inc/utils.h

```diff
@@ -27,14 +45,45 @@
 */
 #define fequals(a, b) (fabs(a - b) < EPSILON) ? (true) : (false)
 
-// Macro that retries a SPI TX/RX if it fails
-#define SPIRTRY(fn, hspi, ...) \
-	for (int i = 0; i < 2; i++) \
+/*!
+  @brief    Carries out a HAL SPI command and blocks the executing task until SPI complete
+  @param    fn          HAL SPI function to execute
+  @param    hspi        SPI bus handle for transaction
+  @param    timeout     Max duration of task blocking state
+  @param    status      U32 status variable that will be set according to the SPI transaction state on completion
+  @param    callingFunc Name of the function that called SPI_TRANSMIT for debugging purposes
+*/
+#define SPI_TRANSMIT(fn, hspi, timeout, status, callingFunc, ...) \
+    for (uint32_t i = 0; i < NUM_SPI_RETRY; i++) \
+    { \
+        if (fn(hspi, __VA_ARGS__) != HAL_OK) \
+        { \
+            Debug("SPI transmission failed to start in function: %s! - Attempt %lu!\n", #callingFunc, i+1); \
+            HAL_SPI_Abort(hspi); \
+            continue; \
+        } \
+        if (xTaskNotifyWait(pdFALSE, ULONG_MAX, status, timeout) != pdPASS) \
```

Comment:
Not sure I like the `pdFalse` and `ULONG_MAX` here. Maybe a #define `CLEAR_ALL_ON_EXIT`?